service_configuration_lib/spark_config.py

```diff
@@ -221,6 +221,8 @@ def _append_sql_shuffle_partitions_conf(spark_opts: Dict[str, str]) -> Dict[str,
         return spark_opts
 
     num_partitions = 2 * (
+        int(spark_opts.get('spark.dynamicAllocation.maxExecutors', 0)) *
```

Comment:
What is the default value of `spark.dynamicAllocation.maxExecutors`?

Reply:
I would just ignore it and let other configs define that value. If all of them are `0`, may be set as some default value (if not already defined somewhere), say `1024`?

Reply:
Note: Make sure of `1024` is added in calculation, it is not picked in the `max` value.

Reply:
I think a reasonable value can be 2 * initialExecutors * spark.executor.cores